### ISSUE TITLE
forward-sexp and backward-sexp are incoherent

### ISSUE THREAD

**Issue by DamienCassou** [mooz/js2-mode#574]
Given this content

```js
values
    .filter((a) => a)
    .filter((a) => a);
```
in a `js2-mode` buffer, `forward-sexp` seems to work fine but `backward-sexp` goes back to the beginning of the first line when invoked with point at the end of second line.

More information at https://github.com/AmaiKinono/puni/issues/9.

**Comment by dgutov** [comment]
Yes, there is certain imbalance in their working. It has been that way for basically ever, I think.

I'll look into this when I have the time. If someone would like to propose a patch, though -- welcome.
